### PR TITLE
[develop] Add bad URL suffix pull request check

### PR DESCRIPTION
**Pull request by hehe7318**
### Description of changes
* Add bad URL suffix pull request check

### Tests
* I added some test URLs(e.g. # This is a test URL: https://example.c2s.ic.gov) in several files and create a PR. It can correctly detect.

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -221,3 +221,12 @@ jobs:
...
           ignore_paths: tests
           ignore_names: user_data.sh
+  url-suffix-check:
+    name: URL Suffix Check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run URL Suffix Detection Script
+      run: |
+        chmod +x ./util/detect_bad_urls.sh
```
Comment (gmarciani) [inline .github/workflows/ci.yml]: No need for custom logic. See https://github.com/aws/aws-parallelcluster-cookbook/blob/develop/.github/workflows/unsafe-patterns-checker.yml

File: util/detect_bad_urls.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+EXIT_STATUS=0
+
+echo "Scanning for disallowed URL suffixes..."
+
+for FILE_TYPE in $FILE_TYPES; do
+  FILES=$(git ls-files -- $FILE_TYPE)
+
+  for FILE in $FILES; do
+    for PATTERN in "${DISALLOWED_PATTERNS[@]}"; do
+      if grep -E -q "$PATTERN" "$FILE"; then
```
Comment (gmarciani) [inline util/detect_bad_urls.sh]: With this logic you're scanning the whole repo, but the checker should scan only the diff of the PR.

**Review by gmarciani — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (hehe7318) [reply .github/workflows/ci.yml]: Done

File: util/detect_bad_urls.sh (hunk 2 above)
Comment (hehe7318) [reply util/detect_bad_urls.sh]: Done

**Review by gmarciani — APPROVED** [review]

**Pull request merged**